time/src/main/java/net/jqwik/time/DefaultDateArbitrary.java

```diff
@@ -8,9 +8,147 @@
 
 public class DefaultDateArbitrary extends ArbitraryDecorator<LocalDate> implements DateArbitrary {
 
+	private LocalDate dateMin = LocalDate.MIN;
+	private LocalDate dateMax = LocalDate.MAX;
+	private Year yearMin = Year.of(LocalDate.MIN.getYear());
+	private Year yearMax = Year.of(LocalDate.MAX.getYear());
+	private Month monthMin = Month.JANUARY;
+	private Month monthMax = Month.DECEMBER;
+	private Month[] allowedMonths = new Month[]{Month.JANUARY, Month.FEBRUARY, Month.MARCH, Month.APRIL, Month.MAY, Month.JUNE, Month.JULY, Month.AUGUST, Month.SEPTEMBER, Month.OCTOBER, Month.NOVEMBER, Month.DECEMBER};
+	private int dayOfMonthMin = 1;
+	private int dayOfMonthMax = 31;
+	private DayOfWeek[] allowedDayOfWeeks = new DayOfWeek[]{DayOfWeek.MONDAY, DayOfWeek.TUESDAY, DayOfWeek.WEDNESDAY, DayOfWeek.THURSDAY, DayOfWeek.FRIDAY, DayOfWeek.SATURDAY, DayOfWeek.SUNDAY};
+
 	@Override
 	protected Arbitrary<LocalDate> arbitrary() {
-		return Arbitraries.just(LocalDate.of(2020, 12, 9));
+		optimizeRuntime();
+		Arbitrary<Year> year = generateYears();
+		Arbitrary<Month> month = generateMonths();
+		Arbitrary<Integer> dayOfMonth = generateDayOfMonths();
+		return Combinators.combine(year, month, dayOfMonth)
+						  .as(this::generateValidDateFromValues)
+						  .filter(v -> v != null && !v.isBefore(dateMin) && !v.isAfter(dateMax) && isInAllowedDayOfWeeks(v.getDayOfWeek()));
```

Comment:
Heavy filtering is usually bad for exhaustive generation, edge case generation and shrinking. I wonder if tuning year, month and dayOfMonth generation could get rid of some of the filter parts.

Reply:
The `optimizeRuntime` method makes it possible to filter as few dates as possible.

I put the other methods in `generateDateFromValues`. Is it better now?

Reply:
Maybe `optimizeRuntime` is just a name I don't understand. It looks to me like some of the things therein should either be done when setting the values or be checked when setting the values for inconsistency - like minYear being greater than maxYear.